dpdata/abacus/scf.py

```diff
@@ -364,32 +527,160 @@ def make_unlabeled_stru(
     numerical_orbital=None,
     numerical_descriptor=None,
     mass=None,
+    move=None,
+    velocity=None,
+    mag=None,
+    angle1=None,
+    angle2=None,
+    sc=None,
+    lambda_=None,
+    link_file=False,
+    dest_dir=None,
+    **kwargs,
 ):
+    """
+    Make an unlabeled STRU file from a dictionary.
+    
+    Parameters
+    ----------
+    data : dict
+        System data
+    frame_idx : int
+        The index of the frame to dump
+    pp_file : list of string or dict, optional
+        List of pseudo potential files, or a dictionary of pseudo potential files for each atomnames
+    numerical_orbital : list of string or dict, optional
+        List of orbital files, or a dictionary of orbital files for each atomnames
+    numerical_descriptor : str, optional
+        numerical descriptor file
+    mass : list of float, optional
+        List of atomic masses
+    move : list of list of bool, optional
+        List of the move flag of each xyz direction of each atom
+    velocity : list of list of float, optional
+        List of the velocity of each xyz direction of each atom
+    mag : list of (list of float or float), optional
+        List of the magnetic moment of each atom, can be a list of three floats or one float
+    angle1 : list of float, optional
+        List of the angle1 of each atom. For noncollinear calculation, it is the angle between the magnetic moment and the z-axis.
+    angle2 : list of float, optional
+        List of the angle2 of each atom. For noncollinear calculation, it is the angle between the projection of magnetic moment on xy plane and the x-axis.
+    sc : list of (bool or list of 3 bool), optional
+        List of the spin constraint flag of each atom. Each element can be a bool or a list of three bools.
```

Comment:
inferred by `mag[i] is None`?

Reply:
In abacus, "mag" defines the initial magnetic moment, and also the target magnetic moment when using deltaspin. ABACUS support to only constrain the magnetic moment of partial atoms, which means one can define the initial magnetic but do not constrain it.

Reply:
what would happen if there is `spins` data in `dpdata.System`? please define the behavior and clarify in the doc.

Reply:
When dpdata.System use function `to()` to dump to ABACUS STRU, if has `spins`, then it will be treated as mag; while if `mag` is passed at the same time, then `mag` will be used.

Reply:
please write it in the doc!